frontend/src/lib/components/CommandPalette/commandPaletteLogic.ts

```diff
@@ -0,0 +1,681 @@
+import { kea } from 'kea'
+import { router } from 'kea-router'
+import { commandPaletteLogicType } from 'types/lib/components/CommandPalette/commandPaletteLogicType'
+import Fuse from 'fuse.js'
+import { dashboardsModel } from '~/models/dashboardsModel'
+import { Parser } from 'expr-eval'
+import _ from 'lodash'
+import {
+    CommentOutlined,
+    FundOutlined,
+    RiseOutlined,
+    ContainerOutlined,
+    AimOutlined,
+    CheckOutlined,
+    SyncOutlined,
+    TagOutlined,
+    ClockCircleOutlined,
+    UserOutlined,
+    UsergroupAddOutlined,
+    ExperimentOutlined,
+    SettingOutlined,
+    MessageOutlined,
+    TeamOutlined,
+    LinkOutlined,
+    CalculatorOutlined,
+    FunnelPlotOutlined,
+    GatewayOutlined,
+    InteractionOutlined,
+    MailOutlined,
+    KeyOutlined,
+    VideoCameraOutlined,
+    SendOutlined,
+    LogoutOutlined,
+    PlusOutlined,
+    LineChartOutlined,
+} from '@ant-design/icons'
+import { DashboardType } from '~/types'
+import api from 'lib/api'
+import { appUrlsLogic } from '../AppEditorLink/appUrlsLogic'
+import { copyToClipboard, isURL } from 'lib/utils'
+import { personalAPIKeysLogic } from '../PersonalAPIKeys/personalAPIKeysLogic'
+
+// If CommandExecutor returns CommandFlow, flow will be entered
+export type CommandExecutor = () => CommandFlow | void
+
+export interface CommandResultTemplate {
+    icon: any // any, because Ant Design icons are some weird ForwardRefExoticComponent type
+    display: string
+    synonyms?: string[]
+    prefixApplied?: string
+    executor?: CommandExecutor | true // true means "just clear input"
+    guarantee?: boolean // show result always and first, regardless of fuzzy search
+}
+
+export type CommandResult = CommandResultTemplate & {
+    source: Command | CommandFlow
+    index?: number
+}
+
+export type CommandResolver = (
+    argument?: string,
+    prefixApplied?: string
+) => CommandResultTemplate[] | CommandResultTemplate | null
+
+export interface Command {
+    key: string // Unique command identification key
+    prefixes?: string[] // Command prefixes, e.g. "go to". Prefix-less case is dynamic base command (e.g. Dashboard)
+    resolver: CommandResolver | CommandResultTemplate[] | CommandResultTemplate // Resolver based on arguments (prefix excluded)
+    scope: string
+}
+
+export interface CommandFlow {
+    icon?: any
+    instruction?: string
+    resolver: CommandResolver | CommandResultTemplate[] | CommandResultTemplate
+    scope: string
+}
+
+export type CommandRegistrations = {
+    [commandKey: string]: Command
+}
+
```

Comment:
nit: it's often a good idea to make a new `types.ts` file that all components can just import from. It helps keeps things a bit more organised...

Reply:
That's true, though in this case I feel like these types are neat to have with the logic, as they are sort of the blueprints for it.

Reply:
yep, use your best judgement here :)